Feat(PartyInfo): 파티 상세화면

toastmeister1 opened the pull request:
! 작업이 길어질거 같아서 PR 걸어두고 작업 하겠습니다~

##  파티 상세화면 (PartyInformationActivity)

### # View 구현 (완료)

### # 모델 정의 (완료)

### # Domain과 Data 로직 추가 (예정)

### # Owner인지 아닌지를 먼저 확인하는 기능 (진행중)
처음에 파티 정보를 불러올때 isOwner 필드를 통해서 판단한다.
- Owner인 경우 -> 파티상태 변경TextView 활성화 / UserList에서 강퇴 버튼 활성화 
- Owner가 아닌경우 -> 파티상태 TextView 활성화 / UserList에서 강퇴 버튼 숨김

### # Comment 기능 (진행중)
- [x] **_Comment의 내부에 답글(Child) Comment가 존재함에 따라 고민중_**
    -  S1 -> Comment 타입은 한개로 두고 ViewType으로 뷰들을 다르게 표시
    -  S2 -> ConcatAdapter를 사용하고 Presentation Layer(app)에서는 Parent와 Child Comment로 분리하고
                     각 Comment타입마다 Adapter를 만들어주는 방식 (자세히 봐야하고, 순서를 어떻게 보장해줄지 찾아보는중)

### # Comment 답글쓰기 기능 (예정 - 디자인 수정 제안중)
- [x] 답글쓰기를 누르면 해당 포지션에 답글을 쓰는 창이 띄워지나 구현하는데 시간이 조금 걸릴것으로 판단해 수정 제안중

### # UserList (예정 - 서버개발중)
- [x] 방장인 경우  강퇴가 가능하게끔 해야 한다
    - 강퇴가 성공적으로 되면 UserList 한번 더 불러와 갱신
  
### # SharedElement (진행중)
    - Fragment -> Activity 의 동작은 Playground에서 확인했지만 
       실제로는 Fragment -> Activity로 Party정보를 보내주고 intent를 받은 후에는
       API 콜을 통해 해당 파티 정보를 한 번 더 갱신해줘야 함

toastmeister1 committed:
feat(partyInfo): add Assets

toastmeister1 committed:
refactor(partyInfo): FoodType Sealedclass로 생성

toastmeister1 committed:
fix(partyInfo): Activity이름 변경

toastmeister1 committed:
merge(gradle): hotfix merge

toastmeister1 committed:
feat(partyInfo): partyinformation XML 작업 진행중 (40%..)

- 댓글 입력창 진행중

toastmeister1 committed:
merge(partyInfo): resource PR merge

toastmeister1 committed:
feat(partyInfo): xml 작업 70%

toastmeister1 committed:
Merge branch 'develop' into feat/partyInfo

# Conflicts:
#	DeliBuddy/.idea/misc.xml
#	DeliBuddy/app/src/main/AndroidManifest.xml
#	DeliBuddy/app/src/main/res/drawable/icon_back.xml
#	DeliBuddy/app/src/main/res/drawable/icon_location.xml

toastmeister1 committed:
merge(develop): merge develop branch

toastmeister1 committed:
Merge branch 'develop' of https://github.com/YAPP-19th/Android-Team-1-Frontend into feat/partyInfo

toastmeister1 committed:
fix(feat/partyInfo): foodType 네이밍 컨벤션에 맞게 수정

toastmeister1 committed:
feat(partyInfo): 모델들 선언

toastmeister1 committed:
feat(partyInfo): Comment Adapter 생성

toastmeister1 committed:
feat(partyInfo): CommentAdapter 답글쓰기 리스너 등록 및 구조 개선

- ViewHolder를 다른 파일로 분리
- 가독성 개선

toastmeister1 committed:
2021. 12. 04
- CommentReclycerView 테스트 필요

toastmeister1 committed:
feat(partyInfo): owner아이콘 추가

toastmeister1 committed:
feat(partyInfo): CommunityTab XML 작업

- Community TabLayout은 Viewpager2 + Tablayout으로 구성
- 각 Tab은 Fragment로 구성
- UserList Tab부분은 서버 미구현에 따라 보류
- Comment Tab부분은 완성했으나 데이터구조를 고려해서 변경 구상중

toastmeister1 committed:
feat(partyInfo): 파티 상태 View 추가

- Owner인 경우 -> 파티 상태를 변경할 수 있는 View 표시
- Owner가 아닌경우 -> 파티 상태만을 알려주는 View 표시

jsh-me reviewed (commented):
아직 wip 인가요??

jsh-me commented on DeliBuddy/app/src/main/java/yapp/android1/delibuddy/adapter/CommentAdapter.kt:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    override fun areContentsTheSame(oldItem: Comment, newItem: Comment): Boolean {
+        return oldItem.id == newItem.id
```
```suggestion
        return oldItem == newItem
```

jsh-me commented on DeliBuddy/app/src/main/java/yapp/android1/delibuddy/model/Comment.kt:
```diff
@@ -0,0 +1,25 @@
+package yapp.android1.delibuddy.model
+
+
+class Comment(
```
```suggestion
data class Comment(
```

jsh-me commented on DeliBuddy/app/src/main/java/yapp/android1/delibuddy/ui/partyInformation/fragments/CommentTabFragment.kt:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    private fun initializeRecyclerView() = with(binding) {
+        rvComment.adapter = commentAdapter
```
```suggestion
        rvComment.adapter = CommentAdapter()
```
바로 선언하는게 어떨까요?

toastmeister1 replied:
넵 그게 좋겠네유

jsh-me commented on DeliBuddy/app/src/main/java/yapp/android1/delibuddy/ui/partyInformation/fragments/CommentTabFragment.kt:
```diff
@@ -0,0 +1,52 @@
+
+    private val viewModel = activityViewModels<PartyInformationViewModel>()
+
+    private val commentAdapter = CommentAdapter()
```
불필요한 변수 선언 같아요 !

toastmeister1 commented:
넵 아직 진행중이고 Comment 부분을 어떻게 처리할지 고민좀 하고 있네요!

jihee-dev commented on DeliBuddy/app/src/main/AndroidManifest.xml:
```diff
@@ -2,23 +2,17 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.DeliBuddy">
         <activity
-            android:name=".ui.address.AddressActivity"
```
혹시 이 부분이 빠져도 되는 걸까요?

toastmeister1 replied:
??? 왜 빠져있지..? 아마 빌드 실패가 저것때문인것 같네요
머지 잘못한거 같아요.. 빠른 수정하겠습니다

jihee-dev replied:
넹 좋아요~!

toastmeister1 committed:
fix(partyInfo): Manifest 잘못 머지된 부분 수정

toastmeister1 committed:
fix(partyInfo): 쥐방울 만한 네이밍 수정

toastmeister1 committed:
Merge branch 'develop' of https://github.com/YAPP-19th/Android-Team-1-Frontend into feat/partyInfo

toastmeister1 committed:
merge(develop): 로그인 기능 merge

toastmeister1 committed:
Merge branch 'develop' of https://github.com/YAPP-19th/Android-Team-1-Frontend into feat/partyInfo

toastmeister1 committed:
feat(comments): Comments 표현 구현

- ParentViewHolder내부에 RecyclerView를 사용
- Adapter는 ViewType을 이용해 같은 Adapter를 재사용 해도 되는 구조로 구현

toastmeister1 committed:
feat(comments): Comment 모델 수정 및 추가

toastmeister1 committed:
feat(comments): Comment 추가, 삭제, 조회 기능 추가

toastmeister1 committed:
feat(comments): Comment기능 DI 추가

toastmeister1 committed:
merge(develop): Home(PartyList)화면 merge

toastmeister1 committed:
feat(comments): Comment 조회 기능 구현

- Comment리스트들을 표시하는데만 성공
- 삭제, 작성은 로직만 구현하면 된다 우선순위는 View단에서 어떻게 해야 자연스럽게
 표시를 해줄지를 고민

toastmeister1 committed:
feat(comments): SharedElement기능 구현

toastmeister1 committed:
fix(comments): Icon 마진값 수정

toastmeister1 committed:
Merge remote-tracking branch 'origin/develop' into develop

toastmeister1 committed:
Merge branch 'develop' into feat/partyInfo

toastmeister1 committed:
Merge remote-tracking branch 'origin/develop' into feat/partyInfo

toastmeister1 committed:
feat(comments): merge하면서 날라간 SharedElement 기능 다시 추가

toastmeister1 committed:
feat(partyInfo): add Custom Toolbar

toastmeister1 committed:
refactor(partyInfo): PartyInformation 화면 구현
